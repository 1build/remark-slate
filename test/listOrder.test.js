const plugin = require('../');
const listOrderMdast = require('./listOrder-mdast');

const transform = plugin.transform;

it('It reorders lists when their children of list-items', () => {
  expect(listOrderMdast.children.map(transform)).toMatchSnapshot();
});


it('It works when flattenListItems is true', () => {
  expect(listOrderMdast.children.map(w => transform(w, { flattenListItems: true }))).toMatchSnapshot();
});