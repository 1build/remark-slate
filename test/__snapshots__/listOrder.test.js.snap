// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It reorders lists when their children of list-items 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Bullet 1",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Bullet 2",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "text": "Nested 1 Bullet 1",
                  },
                ],
                "type": "paragraph",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "text": "Nested 1 Bullet 2",
                  },
                ],
                "type": "paragraph",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "Nested 2 Bullet 1",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "type": "list-item",
              },
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "Nested 2 Bullet 2",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "type": "list-item",
              },
            ],
            "type": "bulleted-list",
          },
        ],
        "type": "bulleted-list",
      },
    ],
    "type": "bulleted-list",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Bullet 1",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Bullet 2",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "text": "Nested 1 Bullet 1",
                  },
                ],
                "type": "paragraph",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "text": "Nested 1 Bullet 2",
                  },
                ],
                "type": "paragraph",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "Nested 2 Bullet 1",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "type": "list-item",
              },
              Object {
                "children": Array [
                  Object {
                    "children": Array [
                      Object {
                        "text": "Nested 2 Bullet 2",
                      },
                    ],
                    "type": "paragraph",
                  },
                ],
                "type": "list-item",
              },
            ],
            "type": "numbered-list",
          },
        ],
        "type": "numbered-list",
      },
    ],
    "type": "numbered-list",
  },
]
`;

exports[`It works when flattenListItems is true 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "Bullet 1",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "text": "Bullet 2",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Nested 1 Bullet 1",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "text": "Nested 1 Bullet 2",
              },
            ],
            "type": "list-item",
          },
        ],
        "type": "bulleted-list",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Nested 2 Bullet 1",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "text": "Nested 2 Bullet 2",
              },
            ],
            "type": "list-item",
          },
        ],
        "type": "bulleted-list",
      },
    ],
    "type": "bulleted-list",
  },
  Object {
    "children": Array [
      Object {
        "children": Array [
          Object {
            "text": "Bullet 1",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "text": "Bullet 2",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Nested 1 Bullet 1",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "text": "Nested 1 Bullet 2",
              },
            ],
            "type": "list-item",
          },
        ],
        "type": "numbered-list",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Nested 2 Bullet 1",
              },
            ],
            "type": "list-item",
          },
          Object {
            "children": Array [
              Object {
                "text": "Nested 2 Bullet 2",
              },
            ],
            "type": "list-item",
          },
        ],
        "type": "numbered-list",
      },
    ],
    "type": "numbered-list",
  },
]
`;
